alpine/Dockerfile

```diff
@@ -18,12 +18,19 @@ WORKDIR /opt
 
 COPY sql-requirements.in ./
 
-# `g++` is required for building `psycopg2` but all build dependencies are
-# later removed again to reduce the layer size.
 RUN set -eux \
-    && apk add --no-cache --virtual .build-deps postgresql-dev g++ \
+    # Install build dependencies to make sure we can build packages that don't
+    # provide pre-built wheels from source.
+    && apk add --no-cache --virtual .build-deps g++ \
+    # Install dev libraries for psycopg2. Note that these are needed at runtime.
+    && apk add --no-cache postgresql-dev python3-dev \
+    # Compile the dependencies for base + SQL.
```

Comment:
I would structure it in the following way:

```dockerfile
    && apk add --no-cache --virtual .build-deps g++ postgresql-dev \
    && apk add --no-cache postgresql \
```

We could also update the index once and later remove it but fetching seems pretty fast.

```dockerfile
&& apk update
...
&& rm -rf /var/cache/apk/*
```

Reply:
The difference in size between including the headers or not is 13 MB btw.

Reply:
Ok, thank you - the 13 MB are now shaved off.